src/node_lkl.cc

```diff
@@ -150,3 +150,38 @@ NAN_METHOD(syscall) {
 		}
 	}
 }
+
+NAN_METHOD(parseDirent64) {
+	// Args:
+	// * Buffer buffer
+	// * int nread
+	// * Array result
+	// Parses buffer (which should be a lkl_linux_dirent64*) and fills the
+	// result array with filenames it finds.
+	// Filenames are Buffers that need to be decoded by the caller.
+	char* buf = (char*) node::Buffer::Data(info[0]);
+	int nread = info[1]->IntegerValue();
```

Comment:
If this is negative it will cause an infinite loop, you can use `->Uint32Value()`